src/combinatorics/simple_graph/basic.lean

```diff
@@ -97,14 +102,22 @@ def simple_graph.from_rel {V : Type u} (r : V → V → Prop) : simple_graph V :
   symm := λ a b ⟨hn, hr⟩, ⟨hn.symm, hr.symm⟩,
   loopless := λ a ⟨hn, _⟩, hn rfl }
 
-noncomputable instance {V : Type u} [fintype V] : fintype (simple_graph V) :=
-by { classical, exact fintype.of_injective simple_graph.adj simple_graph.ext }
+/-- Construct the simple graph induced by a given irreflexive relation. It symmetrizes the relation.
+-/
+def simple_graph.from_irrefl_rel (r : α → α → Prop) [is_irrefl α r] : simple_graph α :=
+{ adj := λ a b, r a b ∨ r b a,
+  symm := λ a b, or.symm,
+  loopless := λ a h, h.elim (irrefl _) (irrefl _) }
```

Comment:
It definitely feels wrong to make the relation irreflexive when you don't need to. Having a dedicated constructor is not much to sustain either.

Reply:
why? I don't see a reason to define a construction that sometimes works, but is equal to a construction that always works.

Reply:
I'm thinking we might want to have *neither* constructor. There should be some combinators for relations to make them be symmetric or irreflexive, and these can be plugged into the usual `simple_graph` constructor.

`from_rel_adj` is not being used at the moment, and it's awkward enough that I avoid using it myself. I believe it's mostly there as a hint for how you might create a `simple_graph`. It would be good to include how to symmetrify and/or irreflexivize a relation in the standard way in the `simple_graph` docstring.

Reply:
Okay, I will inline the definition. I like it like that. Seems like we don't have `relation.symm_gen` nor `relation.symm_irrefl_gen`, which would be the combinators you're talking about.